src/docs/user/references/Variables.adoc

```diff
@@ -0,0 +1,487 @@
+
+Variables can come at hand while developing ProActive applications.
+Propagate data between tasks, customize jobs, and store useful debug information are a few examples of
+how variables can ease development. ProActive has basically two groups of variables :
+
+- <<_workflow_variables>>, declared on the XML job definition.
+- <<_dynamic_variables>>, system variables or variables created at task runtime.
+
+[[_workflow_variables]]
+==== Workflow variables
+
+Workflow variables are declared in the XML job definition. Within the definition we can have variables in several
+levels, at job level the variables are visible and shared by all tasks. At task level, variables are visible
+only within the task execution context.
+
+[[_job_variables]]
+===== Job variables
+
+In a workflow you can define *job variables* that are shared and visible by all tasks.
+
+A job variable is defined using the following attributes:
+
+* *name*: the name of the variable
+* *value*: the value of the variable
+* *model*: the type of the value (optional). See section <<_variable_model,Variable Model>>
+
+
+[[_templating]]
+Variables are very useful to use workflows as templates where only a few parameters change for each
+execution. Upon submission you can define variable values either from <<_scheduler_command_line, CLI>> or from
+the Web interfaces.
+
+The following example shows how to define a job variable in the XML schema:
+[source, xml]
+----
+<job ... >
+    <variables>
+        <variable name="variable" value="value" model=""/>
+    </variables>
+    ...
+</job>
+----
+
+*Job variables* can be referenced anywhere in the workflow, including other job variables.
+
+The syntax for referencing a variable is the pattern `${variable_name}` (case-sensitive), for example:
+
+[source, xml]
+----
+<nativeExecutable>
+    <staticCommand value="${variable}" />
+</nativeExecutable>
+----
+
+===== Task variables
+
+Similarly to job variables, *Task variables* can be defined within task scope in the job XML definition.
+Task variables scope is strictly limited to the task definition.
+
+A task variable is defined using the following attributes:
+
+* *name*: the name of the variable
+* *value*: the value of the variable
+* *inherited*: asserts whether the propagated value should be use instead of the *value* on the workflow attribute.
+If _true_, the *value* field can be empty and as a consequence this variable is dynamically propagated from a
+previous task.
+* *model*: the type of the value (optional). See section <<_variable_model,Variable Model>>
+
+When the inherited parameters is true the variable value is propagated by a previous task (See
+<<_inherited_variables>> for more details). Value field can be left empty however it can
```

Comment:
Add "In this case," at the beginning of the sentence "Value field can be left..."

Reply:
ok